convert2rhel/checks.py

```diff
@@ -273,26 +279,26 @@ def check_custom_repos_are_valid():
 
 
 def ensure_compatibility_of_kmods():
-    """Ensure if the host kernel modules are compatible with RHEL."""
+    """Ensure that the host kernel modules are compatible with RHEL.
+
+    :raises SystemExit: Interrupts the conversion because some kernel modules are not supported in RHEL.
+    """
     host_kmods = get_loaded_kmods()
     rhel_supported_kmods = get_rhel_supported_kmods()
     unsupported_kmods = get_unsupported_kmods(host_kmods, rhel_supported_kmods)
-    if unsupported_kmods:
-        not_supported_kmods = "\n".join(
-            map(
-                lambda kmod: "/lib/modules/{kver}/{kmod}".format(kver=system_info.booted_kernel, kmod=kmod),
-                unsupported_kmods,
-            )
-        )
+
+    # Validate the best case first. If we don't have any unsupported_kmods, this means
+    # that everything is compatible and good to go.
+    if not unsupported_kmods:
+        logger.debug("All loaded kernel modules are available in RHEL.")
+    else:
         logger.critical(
-            (
-                "The following kernel modules are not supported in RHEL:\n{kmods}\n"
-                "Make sure you have updated the kernel to the latest available version and rebooted the system. "
-                "Remove the unsupported modules and run convert2rhel again to continue with the conversion."
-            ).format(kmods=not_supported_kmods, system=system_info.name)
+            "The following loaded kernel modules are not available in RHEL:\n{0}\n"
+            "Prevent the modules from loading by following {1}"
```

Comment:
```suggestion
            "First, make sure you have updated the kernel to the latest available version and rebooted the system.\n"
            "If this message appears again after doing the above, prevent the modules from loading by following {1}"
```
@abadger, @SpyTec, @r0x0d, as we discussed on the Cabal session, without having the kernel currency check available on OL 8, a message notifying the user that the kernel needs to be up-to-date is useful.
I was thinking whether to go into more detail, saying that the loaded kernel should be the latest available for the minor version of the system they are converting but:
1. the message would not be concise
2. it would be difficult to phrase in a way that works also for a distro without minor versions (Stream)

Reply:
done.